src/lib/y2network/connection_config/ip_config.rb

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+module Y2Network
+  module ConnectionConfig
+    class IPConfig
+      # @return [IPAddress] IP address
+      attr_accessor :address
+      # @return [Symbol] Address scope (:global, :site, :link or :host)
+      attr_accessor :scope
+      # @return [String,nil] Address label
+      attr_accessor :label
+      # @return [IPAddress,nil] Remote IP address of a point to point connection
+      attr_accessor :remote_address
+      # @return [IPAddress,nil] Broadcast address
+      attr_accessor :broadcast
+
+      # Constructor
+      #
+      # @param address [IPAddress]
+      # @param id      [Symbol,String] ID (needed for sysconfig backend in order to write suffixes in
+      #   ifcfg-* files)
```

Comment:
cannot we simply generate ids ourself? is only requirement to be unique or something else?

Reply:
It is slightly tricky (actually, I would rather prefer to drop this `id`). Think about this case (these entries are written by the user):

```ini
IPADDR_SOMEID='192.168.122.1/24'
LABEL_SOMEID='my-label'
```

In the current version, this "alias" is shown in the interface, but it is not handled properly. This is the result when the settings are written:

```ini
IPADDR_SOMEID='192.168.122.1/24'
LABEL_SOMEID='my-label'
IPADDR_1='192.168.122.1/24'
LABEL_1='my-label'
```

Nice, uh? So the idea of the `id` is to keep the `_SOMEID` part, so we do not create duplicated entries. However, I was thinking about dropping it and just rewriting the whole collection like:

```ini
IPADDR_1='192.168.122.1/24'
LABEL_1='my-label'
```
We would drop the "user defined" suffix, but I think it is acceptable. What's your opinion?

Reply:
not sure as I think some customers will complain if they use some meaningful suffix. We will need that generator if we want to hide this suffix stuff from user, but we can generate it elsewhere and not in this code.

Reply:
In such a case, I fear that we should keep the `id` as it is by now (read from the ifcfg-* files).